### PR TITLE
test(provider/xai): rewrite tests to use fixture-based pattern

### PR DESCRIPTION
**Pull request by dancer**
## background

part of #12270 — migrating provider tests to use real API fixtures instead of hand-constructed inline responses

## summary

- record text and tool call fixtures from grok-3-mini (both doGenerate JSON and doStream chunks)
- add `prepareJsonFixtureResponse` and `prepareChunksFixtureResponse` helpers
- add fixture-based describe blocks for doGenerate (text, tool call) and doStream (text, tool call)
- convert all `toStrictEqual` assertions to `toMatchInlineSnapshot()`
- remove old `prepareJsonResponse`, `prepareStreamResponse`, and `prepareReasoningResponse` helpers
- replace inline response mocking with fixture loading for request-format tests
- keep inline responses for edge cases (citations, missing usage, error handling)
- all fixtures include reasoning_content from grok-3-mini, covering reasoning parsing in text/tool-call snapshots

## verification

<details>
<summary>tests pass</summary>

```
$ pnpm test

 ✓ src/convert-xai-chat-usage.test.ts
 ✓ src/convert-to-xai-chat-messages.test.ts
 ✓ src/responses/convert-to-xai-responses-input.test.ts
 ✓ src/responses/xai-responses-prepare-tools.test.ts
 ✓ src/xai-provider.test.ts
 ✓ src/xai-chat-language-model.test.ts
 ✓ src/responses/xai-responses-language-model.test.ts

 Test Files  7 passed (7)
      Tests  157 passed (157)
```

</details>

## checklist

- [x] tests have been added / updated (for bug fixes / features)
- [ ] documentation has been added / updated (for bug fixes / features)
- [ ] a _patch_ changeset for relevant packages has been added (run `pnpm changeset` in root)
- [x] i have reviewed this pull request (self-review)

## related issues

part of #12270

### CODE REVIEW

File: packages/xai/src/__fixtures__/xai-tool-call.chunks.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+{"id":"de9d896d-e946-b3a7-bb14-75ab33326930","object":"chat.completion.chunk","created":1770774064,"model":"grok-3-mini","choices":[{"index":0,"delta":{"reasoning_content":" the"}}],"system_fingerprint":"fp_2a885414fb"}
+{"id":"de9d896d-e946-b3a7-bb14-75ab33326930","object":"chat.completion.chunk","created":1770774064,"model":"grok-3-mini","choices":[{"index":0,"delta":{"reasoning_content":" user"}}],"system_fingerprint":"fp_2a885414fb"}
+{"id":"de9d896d-e946-b3a7-bb14-75ab33326930","object":"chat.completion.chunk","created":1770774064,"model":"grok-3-mini","choices":[{"index":0,"delta":{"reasoning_content":" is"}}],"system_fingerprint":"fp_2a885414fb"}
+{"id":"de9d896d-e946-b3a7-bb14-75ab33326930","object":"chat.completion.chunk","created":1770774066,"model":"grok-3-mini","choices":[{"index":0,"delta":{"tool_calls":[{"id":"call_55117580","function":{"name":"weather","arguments":"{\"location\":\"San Francisco\"}"},"index":0,"type":"function"}]}}],"system_fingerprint":"fp_2a885414fb"}
```
Comment (lgrammel) [inline packages/xai/src/__fixtures__/xai-tool-call.chunks.txt:6]: xai does not stream tool calls?

**Review by lgrammel — APPROVED** [review]
